api/rosetta/data_transaction_integration_test.go

```diff
@@ -0,0 +1,554 @@
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// +build integration
+
+package rosetta_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/configuration"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/meta"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+func TestGetTransaction(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	var (
+		firstHeader      = knownHeaders(44)
+		multipleTxHeader = knownHeaders(165)
+		lastHeader       = knownHeaders(181)
+	)
+
+	const (
+		firstTx = "d5c18baf6c8d11f0693e71dbb951c4856d4f25a456f4d5285a75fd73af39161c"
+
+		// two transactions in a single block
+		firstOfTwoTx  = "23c486cfd54bca7138b519203322327bf46e43a780a237d1c5bb0a82f0a06c1d"
+		secondOfTwoTx = "3d6922d6c6fd161a76cec23b11067f22cac6409a49b28b905989db64f5cb05a5"
+
+		lastTx = "780bafaf4721ca4270986ea51e659951a8912c2eb99fb1bfedeb753b023cd4d9"
+	)
+
+	tests := []struct {
+		name string
+
+		request              rosetta.TransactionRequest
+		validateTransactions transactionValidationFn
+	}{
+		{
+			name:                 "some cherry picked transaction",
+			request:              requestTransaction(firstHeader, firstTx),
+			validateTransactions: validateTransfer(t, firstTx, "754aed9de6197641", "631e88ae7f1d7c20", 1),
+		},
+		{
+			name:                 "first in a block with multiple",
+			request:              requestTransaction(multipleTxHeader, firstOfTwoTx),
+			validateTransactions: validateTransfer(t, firstOfTwoTx, "8c5303eaa26202d6", "72157877737ce077", 100_00000000),
+		},
+		{
+			name:                 "second in a block with multiple",
+			request:              requestTransaction(multipleTxHeader, secondOfTwoTx),
+			validateTransactions: validateTransfer(t, secondOfTwoTx, "89c61aa64423504c", "82ec283f88a62e65", 1),
+		},
+		{
+			name:                 "last transaction recorded",
+			request:              requestTransaction(lastHeader, lastTx),
+			validateTransactions: validateTransfer(t, lastTx, "668b91e2995c2eba", "89c61aa64423504c", 1),
+		},
+		{
+			name: "lookup using height and transaction hash",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID: identifier.Block{
+					Index: 165,
+				},
+				TransactionID: identifier.Transaction{
+					Hash: secondOfTwoTx,
+				},
+			},
+			validateTransactions: validateTransfer(t, secondOfTwoTx, "89c61aa64423504c", "82ec283f88a62e65", 1),
+		},
+
+		/* TODO: this is an invalid test case - looking up a transaction that does not exist in a block or does not move funds
```

Comment:
Dilemma whether to include this test case here - to document the current behaviour, or to perhaps include the test cases of the expected behaviour, even if they fail in the meantime..?

Reply:
Not sure either. For sure we shouldn't have tests that don't pass, as it would break the CI, but having a comment about it sure is a good idea.

Reply:
I think it's fine to put a `//TODO` with a link to an issue to enable it once the bug is fixed.

However, please always use `//` for comments, even multi-line. That's what's supported by IDEs generally, so it's easier to remove comments with one button combo.